Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -378,6 +388,49 @@ EncodedDataStatus ImageDecoderCG::encodedDataStatus() const
     return m_encodedDataStatus;
 }
 
+bool ImageDecoderCG::hasHDRGainMapContent() const
+{
+#if HAVE(SUPPORT_HDR_DISPLAY)
+    auto properties = adoptCF(CGImageSourceCopyProperties(m_nativeDecoder.get(), createImageSourceMetadataOptions().get()));
+    if (!properties)
+        return false;
+
+    auto fileContentsProperties = (CFDictionaryRef)CFDictionaryGetValue(properties.get(), kCGImagePropertyFileContentsDictionary);
+    if (!fileContentsProperties)
+        return false;
+
+    auto imagesInfoArray = (CFArrayRef)CFDictionaryGetValue(fileContentsProperties, kCGImagePropertyImages);
+    if (!imagesInfoArray || !CFArrayGetCount(imagesInfoArray))
+        return false;
+
+    auto imageInfo = (CFDictionaryRef)CFArrayGetValueAtIndex(imagesInfoArray, 0);
+    if (!imageInfo)
+        return false;
+
+    auto auxiliaryDataArray = (CFArrayRef)CFDictionaryGetValue(imageInfo, kCGImagePropertyAuxiliaryData);
+    if (!auxiliaryDataArray)
+        return false;
+
+    CFIndex count = CFArrayGetCount(auxiliaryDataArray);
+    for (CFIndex index = 0; index < count; ++index) {
+        auto auxiliaryData = (CFDictionaryRef)CFArrayGetValueAtIndex(auxiliaryDataArray, index);
+        if (!auxiliaryData)
+            continue;
+
+        auto type = (CFStringRef)CFDictionaryGetValue(auxiliaryData, kCGImagePropertyAuxiliaryDataType);
+        if (!type)
+            continue;
+
+        if (kCFCompareEqualTo == CFStringCompare(type, kCGImageAuxiliaryDataTypeHDRGainMap, 0))
```

Comment:
Generally we put the `kCFCompareEqualTo` on the right.

Reply:
Fixed.